### DISCUSSION TITLE
TypeScript compilation error after latest upgrade

### DISCUSSION THREAD

**Discussion by mastermunj** [adonisjs/core#1847] [Help] [answered]
Hi,

I am following the process given [here](https://preview.adonisjs.com/blog/october-2020-release) for latest upgrade.

When I run `npm run build`, I get following error:

```ts
node_modules/@adonisjs/lucid/build/adonis-typings/querybuilder.d.ts:473:37 - error TS1005: ',' expected.

473         match(...blocks: ([condition: any, callback: (query: this) => any] | ((query: this) => any))[]): this;
                                        ~
node_modules/@adonisjs/lucid/build/adonis-typings/querybuilder.d.ts:473:52 - error TS1005: ',' expected.

473         match(...blocks: ([condition: any, callback: (query: this) => any] | ((query: this) => any))[]): this;
```

I couldn't figure out exactly why this error comes up as node_modules is already in exclude list of `tsconfig.json`.

Am I doing something wrong in the upgrade process?

**Comment by mastermunj** [comment, marked as answer]
Finally, it's resolved.

I was on TypeScript 3.9.7 earlier. After upgrading TypeScript to 4.0.5, the error resolved.

Although I couldn't understand the cause of the error.

**Reply by julians300** [reply]
Thanks. This saved me some time
